### PR TITLE
Support ILIKE and NOT ILIKE operator

### LINKED ISSUE #314: `ILIKE` and `NOT ILIKE` support
`ILIKE` and `NOT ILIKE` are case-insensitive `LIKE` and `NOT LIKE`.

ref https://github.com/gluesql/gluesql/pull/252

### PR DESCRIPTION
**Pull request by maruoovv**
Support ILIKE, NOT ILIKE operator.
These operators are case-insensitive LIKE and NOT LIKE operator.
For adding this, i had upgrade sqlparser version to 0.10.

resolve #314

### CODE REVIEW

**Comment by panarch** [comment]
Test failed because of `sqlpaser-rs` version, there's a ast change between sqlparser 0.9 and 0.10 version, so I would recommend to keep v0.9.

File: src/data/string_ext.rs [hunk 1]
```diff
@@ -7,16 +7,28 @@ pub enum StringExtError {
...
 impl StringExt for String {
-    fn like(&self, pattern: &str) -> Result<bool> {
+    fn like(&self, pattern: &str, case_sensitive: bool) -> Result<bool> {
+        let (match_string, match_pattern) = match case_sensitive {
+            true => (self.to_string(), pattern.to_string()),
+            false => {
+                let lowercase_string = self.to_lowercase();
+                let lowercase_pattern = pattern.to_lowercase();
+
+                (lowercase_string, lowercase_pattern)
```
Comment (panarch) [inline src/data/string_ext.rs:21]: 👍 👍 👍

**Comment by panarch** [comment]
Currently, Both `TRIM` changes and `ILIKE` changes show all together.
It looks like the issue of pulling the main branch.

Can you try merging by these commands?
You may need to `git reset HEAD~1` to remove top one.
```
$ git remote add upstream git@github.com:gluesql/gluesql.git
$ git pull upstream main
```

**Comment by maruoovv** [comment]
> Currently, Both `TRIM` changes and `ILIKE` changes show all together.
> It looks like the issue of pulling the main branch.
> 
> Can you try merging by these commands?
> You may need to `git reset HEAD~1` to remove top one.
> 
> ```
> $ git remote add upstream git@github.com:gluesql/gluesql.git
> $ git pull upstream main
> ```

i'm reset and pull upstream main, can you check if it looks right?

**Comment by panarch** [comment]
> > Currently, Both `TRIM` changes and `ILIKE` changes show all together.
> > It looks like the issue of pulling the main branch.
> > Can you try merging by these commands?
> > You may need to `git reset HEAD~1` to remove top one.
> > ```
> > $ git remote add upstream git@github.com:gluesql/gluesql.git
> > $ git pull upstream main
> > ```
> 
> i'm reset and pull upstream main, can you check if it looks right?

Perfect, thanks!

**Review by panarch — APPROVED** [review]
Looks all good, thanks!

**Pull request merged**